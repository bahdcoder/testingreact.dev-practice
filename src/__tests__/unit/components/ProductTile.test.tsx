import React from 'react'
import { axe } from 'jest-axe'
import { render } from '@testing-library/react'
import ProductTile from '../../../components/ProductTile'

describe('The <ProductTile /> component', () => {
  const defaultProduct = {
    id: 12,
    name: 'Example product name',
    image: '/image.png',
    price: 'from $12.99',
    brand: 'Adidas',
    createdAt: '2020-02-11 00:00:00',
    isActive: true,
  }

  const setupProductTile = (props = defaultProduct) =>
    render(<ProductTile {...(props as any)} />)

  it('renders a product tile with name, image and price', () => {
    const { asFragment, getByText } = setupProductTile()

    expect(getByText(defaultProduct.name)).toBeInTheDocument()
    expect(getByText(defaultProduct.price)).toBeInTheDocument()

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders a product tile with name and price only', () => {
    const { asFragment, getByText, queryByTestId } = setupProductTile({
      ...defaultProduct,
      image: undefined,
    })

    expect(queryByTestId('ProductTileImage')).not.toBeInTheDocument()

    expect(getByText(defaultProduct.name)).toBeInTheDocument()
    expect(getByText(defaultProduct.price)).toBeInTheDocument()

    expect(asFragment()).toMatchSnapshot()
  })

  it('has no accessibility violations', async () => {
    const { container } = setupProductTile()

    expect(await axe(container)).toHaveNoViolations()
  })
})
