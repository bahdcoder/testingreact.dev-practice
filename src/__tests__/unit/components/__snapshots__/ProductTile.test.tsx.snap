// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <ProductTile /> component renders a product tile with name and price only 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iPShpD"
    data-testid="ProductTile"
  >
    <div
      class="sc-AxirZ fUTalh"
    />
    <div
      class="sc-AxgMl dtvmCV"
    >
      <div
        class="sc-AxhCb iMwhLL"
      >
        Example product name
      </div>
      <div
        class="sc-AxhUy dqKFMC"
      >
        from $12.99
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`The <ProductTile /> component renders a product tile with name, image and price 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iPShpD"
    data-testid="ProductTile"
  >
    <div
      class="sc-AxirZ fUTalh"
    >
      <img
        alt="Example product name"
        class="sc-AxiKw cShrNR"
        data-testid="ProductTileImage"
        src="http://localhost:3443/image.png"
      />
    </div>
    <div
      class="sc-AxgMl dtvmCV"
    >
      <div
        class="sc-AxhCb iMwhLL"
      >
        Example product name
      </div>
      <div
        class="sc-AxhUy dqKFMC"
      >
        from $12.99
      </div>
    </div>
  </div>
</DocumentFragment>
`;
